File: typescript/src/renderer.ts
// import { Shape } from './shapes/shape.js";

function newRenderer(object: Shape) {
    return {
        draw() {
            const area: number = object.computeArea()
            console.log("Shape drawn\n" + "Its area is " + area)
        }
    }
}

export { newRenderer }

File: java/src/main/java/edu/cmu/cs/cs214/lab02/Renderer.java
package edu.cmu.cs.cs214.lab02;

import edu.cmu.cs.cs214.lab02.shapes.*;

public class Renderer {
    private Shape rectangle;
    
    Renderer(Shape rectangle) {
        this.rectangle = rectangle;
    }

    void draw() {
        double area = this.rectangle.getArea();

        // assume implementation

        System.out.println("Shape printed\n" + "Its area is " + area);
    }
}
